src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/panels/SettingsPanelNotifications.tsx

```diff
@@ -2,8 +2,219 @@
-  return <div>Set notifications content</div>;
+import { SubscriberRow } from "knex/types/tables";
+import { SubscriberEmailPreferencesOutput } from "../../../../../../../../db/tables/subscriber_email_preferences";
+import { Session } from "next-auth";
+import { useL10n } from "../../../../../../../hooks/l10n";
+import { useSession } from "next-auth/react";
+import { useRouter } from "next/navigation";
+import { hasPremium } from "../../../../../../../functions/universal/user";
+import { createContext, useState } from "react";
+import { RadioGroupState, useRadioGroupState } from "react-stately";
+import {
+  EmailUpdateCommOptionRequest,
+  EmailUpdateCommTypeOfOptions,
+} from "../../../../../../../api/v1/user/update-comm-option/route";
+import { useTelemetry } from "../../../../../../../hooks/useTelemetry";
+import { useRadioGroup } from "react-aria";
+import { VisuallyHidden } from "../../../../../../../components/server/VisuallyHidden";
+import { Button } from "../../../../../../../components/client/Button";
+import styles from "./SettingsPanelNotifications.module.scss";
+import { SwitchInput } from "../../../../../../../components/client/SwitchInput";
+import { RadioInput } from "../../../../../../../components/client/RadioInput";
+
+export type SettingsPanelNotificationsProps = {
+  data: SubscriberEmailPreferencesOutput;
+  subscriber: SubscriberRow;
+  user: Session["user"];
+};
+
+export type NotificationSettingsProps = {
+  user: Session["user"];
+  subscriber: SubscriberRow;
+  data: SubscriberEmailPreferencesOutput;
+};
+
+const AlertAddressContext = createContext<RadioGroupState | null>(null);
+
+export const NotificationsSettings = (props: NotificationSettingsProps) => {
+  const l10n = useL10n();
+  const session = useSession();
+  const router = useRouter();
+
+  const breachAlertsEmailsAllowed = props.subscriber.all_emails_to_primary;
+
+  // Extract monthly report preference from the right column
+  const monitorReportAllowed = hasPremium(props.user)
+    ? props.data.monthly_monitor_report
+    : props.data.monthly_monitor_report_free;
+
+  const defaultActivateAlertEmail =
+    typeof breachAlertsEmailsAllowed === "boolean";
+  const [activateAlertEmail, setActivateAlertEmail] = useState<boolean>(
+    defaultActivateAlertEmail,
+  );
+
+  const [activateMonthlyMonitorReport, setActivateMonthlyMonitorReport] =
+    useState(monitorReportAllowed);
+
+  const commsValue = () => {
+    switch (breachAlertsEmailsAllowed) {
+      case true:
+        return "primary";
+      case false:
+        return "affected";
+      case null:
+        return "null";
+    }
+  };
+
+  const state = useRadioGroupState({
+    defaultValue: commsValue(),
+    onChange: (newValue) => {
+      const chosenOption = newValue as EmailUpdateCommTypeOfOptions;
+      const body: EmailUpdateCommOptionRequest = {
+        instantBreachAlerts: chosenOption,
+      };
+      void fetch("/api/v1/user/update-comm-option", {
```

Comment:
I thought we had replaced this with a server action... would you mind filing a tech debt ticket to do so?

Reply:
The code here for updating the communication options is taken from the existing `AlertAddresForm` (which will be deprecated as soon the new views are live) — so it seems like this hasn’t been updated, yet. I’ve filed [MNTOR-3839](https://mozilla-hub.atlassian.net/browse/MNTOR-3839).